packages/core/src/api/clipboard/fromClipboard/pasteExtension.ts

```diff
@@ -26,6 +26,24 @@ function defaultPasteHandler({
   prioritizeMarkdownOverHTML: boolean;
   plainTextAsMarkdown: boolean;
 }) {
+  // Special case for code blocks, as they do not support any rich text
+  // formatting, so we force pasting plain text.
+  const isInCodeBlock = editor.transact(
+    (tr) =>
+      tr.selection.$from.parent.type.spec.code &&
+      tr.selection.$to.parent.type.spec.code
+  );
+
+  if (isInCodeBlock) {
+    const data = event.clipboardData?.getData("text/plain");
+
+    if (data) {
+      editor.pasteText(data);
+    }
+
+    return true;
```

Comment:
```suggestion
    const data = event.clipboardData?.getData("text/plain");

    if (data) {
      editor.pasteText(data);
      return true;
    }
```

Sorry, realized this afterward

Reply:
Are you sure about this? Imo if there is no plain text we shouldn't paste anything, since otherwise it'll paste all wonky like in the original issue

Reply:
Honestly not sure, I think both cases are not ideal. I think your solution is better for files though come to think of it, since they should get pasted regardless.